property/collector.go

```diff
@@ -164,6 +164,11 @@ func (p *Collector) Retrieve(ctx context.Context, objs []types.ManagedObjectRefe
 		return err
 	}
 
+	if d, ok := dst.(*[]types.ObjectContent); ok {
+		*d = res.Returnval
```

Comment:
`dst interface{}`, if the type is a pointer to `[]types.ObjectContent` we just set dst to that slice and skip the call to `mo.LoadRetrievePropertiesResponse`